kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/V3ResourcesModule.java

```diff
@@ -87,4 +111,50 @@ public void dispose(ExecutorService executorService) {
       }
     }
   }
+
+  private static final class StreamingResponseIdleTimeExecutorFactory
+      implements Factory<ExecutorService> {
+
+    @Override
+    public ExecutorService provide() {
+      return Executors.newCachedThreadPool(
```

Comment:
Why a cached thread pool, rather than a fixed thread pool?

https://www.baeldung.com/java-executors-cached-fixed-threadpool

In particular 
"As a result, fixed thread pools are better suited for tasks with unpredictable execution times." - which is true here I think, our tasks will have unpredictable times

 "cached pool starts with zero threads and can potentially grow to have Integer.MAX_VALUE" - and while I think that's OK, as we shouldn't allow that many requests in parallel, it makes me a little nervous in case we are forgetting to clear up somewhere and leak them - blowing up the JVM is worse than stopping Produce due to lack of threads being available in a fixed size pool.

Reply:
https://javadoc.scijava.org/Guava/com/google/common/util/concurrent/SimpleTimeLimiter.html#create-java.util.concurrent.ExecutorService-

> Warning: using a bounded executor may be counterproductive! If the thread pool fills up, any time callers spend waiting for a thread may count toward their time limit, and in this case the call may even time out before the target method is ever invoked.

In reality, this thread pool is limited to the number of parallel streams, which is limited to the number of request handling threads (since `callWithTimeout` will be invoked from a request handling thread). We could use the same configuration as the request handling thread pool, not sure if necessary though. I will do it if you two think it still a problem.

Reply:
" If the thread pool fills up" - then we have another problem I think - better for it to fill up than to grow beyond what we expect.

I'd strongly prefer a size bounded thread pool. @dimitarndimitrov what do you think?

Reply:
> Warning: using a bounded executor may be counterproductive! If the thread pool fills up, any time callers spend waiting for a thread may count toward their time limit, and in this case the call may even time out before the target method is ever invoked.

This assumes that the bounded executor enqueues tasks. Not the case for the suggestion above.
It's a moot point anyway, as we don't really care whether we'll count the timeout correctly, if we are past the suggested boundary.

> In reality, this thread pool is limited to the number of parallel streams, which is limited to the number of request handling threads (since callWithTimeout will be invoked from a request handling thread). We could use the same configuration as the request handling thread pool, not sure if necessary though. I will do it if you two think it still a problem.

Right, and with our config that's an even tighter bound than the one calculated with the configured idle time and the stream rate limit. I would prefer to use the min of these, but I realize this is a hotfix, so just putting something saner than `Integer.MAX_VALUE` is good by me.